Ember/app/services/user-service.js

```diff
@@ -0,0 +1,73 @@
+import Service from '@ember/service';
+import {inject as service} from '@ember/service';
+import { reject } from 'rsvp';
+
+export default Service.extend({
+    ajax: service(),
+
+    register(firstName, lastName, email, phoneNumber, userType, password) {
+        return new Promise((resolve, reject) => {
```

Comment:
ajax requesti svakako vraćaju promisu, tako da nema potrebe da explicitno praviš svoju promisu pa je resolve-aš u then-u. samo uradi `return this.get('ajax').get(....);`